drud/ddev
# ddev rm, describe, and stop are now reliable even when a site directory is missing; fixes #459

## The Problem/Issue/Bug:
ddev rm, describe, and stop were unable to remove sites that were missing their config.yml or sites that did not have directories associated with them. This made it impossible to cleanup corrupted sites pointed out by ddev list.
## How this PR Solves The Problem:
This PR solves the problem by using a site's containers to grab it's site-name and using that site-name to run Cleanup() on the site. It also checks for a missing directory and if the approot is missing it skips `app.Down()` and uses `Cleanup()`.
## Manual Testing Instructions:
Test this PR using the following steps:
- Start a site using `ddev start`
- `rm -r` the site directory
- `ddev list` the site to see that it reports its status as `app directory missing:`
- Use `ddev rm <sitename>` to remove the site 

## Automated Testing Overview:
Tests for `ddev rm` are run in `TestCleanupWithoutCompose`. Tests for `ddev describe` are run in `TestDescribeMissingDirectory`. And tests for `ddev stop` are run in `TestLocalStopMissingDirectory`.

## Related Issue Link(s):
#459 
## Release/Deployment notes:
